### PR TITLE
chore: add github action ci for node 16

### PR DESCRIPTION
**Pull request by kkalavantavanich**
## Description 

Add GitHub Action CI for Node 16 since it has been promoted to Active LTS since October 2021.

## Related Issue

#1402 

## Motivation and Context

Add CI for Node 16

## How Has This Been Tested?

Should pass new CI pipeline.

## Types of changes

- [x] Updated docs / Refactor code / Added a tests case (non-breaking change)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.
- [ ] I have updated the changelog.

### CODE REVIEW

File: .github/workflows/node.js.yml [hunk 1]
```diff
@@ -12,12 +12,18 @@ on:
...
+        node-version: [10.x, 12.x, 14.x, 15.x, 16.x]
+        ts-project: [src/tsconfig.json, src/tsconfig-es6.json]
+
     steps:
       - uses: actions/checkout@v2
       - uses: nuget/setup-nuget@v1
       - uses: actions/setup-node@v1
         with:
-          node-version: 12
+          node-version: ${{ matrix.node-version }}
```
Comment (notaphplover) [inline .github/workflows/node.js.yml]: Why should we test every NodeJS version on windows? Just curious about it

Comment (dcavanagh) [reply .github/workflows/node.js.yml]: Good question. The purpose of this job is to test in the browser.

Comment (dcavanagh) [inline .github/workflows/node.js.yml]:
```suggestion
          node-version: 16
```

**Review by dcavanagh — CHANGES_REQUESTED** [review]
Please make the requested changes

**Pull request closed without merging**